### PR TITLE
Fix HQVM implementation

### PR DESCRIPTION
**Pull request by Gillou68310**
This is relying on https://github.com/mupen64plus/mupen64plus-rsp-hle/pull/73
So don't merge before https://github.com/mupen64plus/mupen64plus-rsp-hle/pull/73 is merged

### CODE REVIEW

**Comment by CrashOveride95** [comment]
This pull iirc finishes HVQM RSP support, as all possible configurations have been tested, 4:1:1 and 4:2:2 compression, and 16bit RGBA and 32bit RGBA framebuffer options.

**Comment by Gillou68310** [comment]
Thanks @CrashOveride95 for providing all test roms!

File: src/hle.c [hunk 1]
```diff
@@ -352,6 +352,7 @@ static ucode_func_t try_normal_task_detection(struct hle_t* hle)
         return &jpeg_decode_OB;
     }
 
+    /* Resident Evil 2 */
```
Comment (Narann) [inline src/hle.c]: Thanks for those comments, we don't have many, but games or public rom names reaching a particular code place is always useful for later review.

**Comment by Gillou68310** [comment]
Rebased on master

File: src/hvqm.c [hunk 2]
```diff
@@ -199,68 +193,96 @@ static int process_info(struct hle_t* hle, uint8_t* base, int16_t* out)
     return 1;
 }
 
-#define SATURATE(x) ((unsigned int) x <= 31 ? x : (x < 0 ? 0 : 31))
-static uint16_t YCbCr_to_BGRA5551(int16_t Y, int16_t Cb, int16_t Cr, uint8_t alpha)
+#define SATURATE8(x) ((unsigned int) x <= 255 ? x : (x < 0 ? 0: 255))
+static struct RGBA YCbCr_to_RGBA(int16_t Y, int16_t Cb, int16_t Cr, uint8_t alpha)
```
Comment (Narann) [inline src/hvqm.c]: Not a review per see, but just a note: This function is always called from a 4 sized loop (itself inside a loop of a loop), I wonder if it wouldn't be more efficient to check `arg.chroma_step_v` and allocate fixed size array before the loops so you can make this function deals with array and do the component maths at once.

If have no idea about performance anyway so just a note.

Comment (Gillou68310) [reply src/hvqm.c]: Well I organized loops to actually have this function to only deal with one pixel at a time :-)
I think it's easier to understand that way but it's just my opinion ofc.

Comment (Narann) [reply src/hvqm.c]: Yes, it's definitely easier to understand, my point was about performance, but I have no clue.

**Pull request merged**